### DISCUSSION TITLE
How do I connect multiple bluetooth devices on linux mint cinnamon?

### DISCUSSION THREAD

**Discussion by DavidicusBellefonte** [blueman-project/blueman#2293] [Q&A]
Hi, I am new to these forums, and to linux entirely. I simply looked for linux once, and found it was super easy to install via a usb flash and I did that and just kind of now am getting into dipping my toes in asking for help on these foruns, I have literally no idea what I am doing, and so please be patient and kind to me, this is a scary post to be making and I'm sorry if I'm already breaking a rule. 

I am just trying to connect multiple bluetooth devices, which I am supposed to be able to do using Bluenan. I can connect my headphones, but not my bose speaker. I want to connect as many as I can to make a home theater easier and cheaper.
I can see my devices, and I can listen to the audio on my headphones and then connect to the bose, but theres no sound. I have to click to connect on them individually, and cannot seem to find out why it won't let me do it. Please please please, can anyone help me?

System:
  Kernel: 5.15.0-97-generic x86_64 bits: 64 compiler: gcc v: 11.4.0 Desktop: Cinnamon 6.0.4
    tk: GTK 3.24.33 wm: muffin vt: 7 dm: LightDM 1.30.0 Distro: Linux Mint 21.3 Virginia
    base: Ubuntu 22.04 jammy
Machine:
  Type: Desktop Mobo: N/A model: JB20B serial: <superuser required> UEFI: American Megatrends LLC.
    v: JB20B001 date: 07/03/2023
Battery:
  Device-1: hidpp_battery_0 model: Logitech Wireless Keyboard serial: <filter>
    charge: 55% (should be ignored) rechargeable: yes status: Discharging
CPU:
  Info: quad core model: Intel Celeron N5095A bits: 64 type: MCP smt: <unsupported> arch: Tremont
    rev: 0 cache: L1: 256 KiB L2: 1.5 MiB L3: 4 MiB
  Speed (MHz): avg: 1172 high: 2198 min/max: 800/2900 cores: 1: 2198 2: 841 3: 850 4: 799
    bogomips: 15974
  Flags: ht lm nx pae sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx
Graphics:
  Device-1: Intel JasperLake [UHD Graphics] driver: i915 v: kernel ports: active: HDMI-A-1
    empty: DP-1,HDMI-A-2,HDMI-A-3 bus-ID: 00:02.0 chip-ID: 8086:4e55 class-ID: 0300
  Display: x11 server: X.Org v: 1.21.1.4 driver: X: loaded: modesetting unloaded: fbdev,vesa
    gpu: i915 display-ID: :0 screens: 1
  Screen-1: 0 s-res: 1920x1080 s-dpi: 96 s-size: 508x285mm (20.0x11.2") s-diag: 582mm (22.9")
  Monitor-1: HDMI-1 mapped: HDMI-A-1 model: LG (GoldStar) 32inch FHD serial: <filter>
    res: 1920x1080 hz: 60 dpi: 70 size: 698x392mm (27.5x15.4") diag: 801mm (31.5") modes:
    max: 1920x1080 min: 640x480
  OpenGL: renderer: Mesa Intel UHD Graphics (JSL) v: 4.6 Mesa 23.2.1-1ubuntu3.1~22.04.2
    direct render: Yes
Audio:
  Device-1: Intel vendor: Conexant Systems driver: snd_hda_intel v: kernel bus-ID: 00:1f.3
    chip-ID: 8086:4dc8 class-ID: 0403
  Sound Server-1: ALSA v: k5.15.0-97-generic running: yes
  Sound Server-2: PulseAudio v: 15.99.1 running: yes
  Sound Server-3: PipeWire v: 0.3.48 running: yes
Network:
  Device-1: Intel Wireless 8265 / 8275 driver: iwlwifi v: kernel pcie: speed: 2.5 GT/s lanes: 1
    bus-ID: 02:00.0 chip-ID: 8086:24fd class-ID: 0280
  IF: wlp2s0 state: up mac: <filter>
  Device-2: Realtek RTL8125 2.5GbE driver: r8169 v: kernel pcie: speed: 5 GT/s lanes: 1
    port: 3000 bus-ID: 03:00.0 chip-ID: 10ec:8125 class-ID: 0200
  IF: enp3s0 state: down mac: <filter>
Bluetooth:
  Device-1: Intel Bluetooth wireless interface type: USB driver: btusb v: 0.8 bus-ID: 1-8:7
    chip-ID: 8087:0a2b class-ID: e001
  Report: hciconfig ID: hci0 rfk-id: 1 state: up address: <filter> bt-v: 2.1 lmp-v: 4.2
    sub-v: 100 hci-v: 4.2 rev: 100
Drives:
  Local Storage: total: 476.94 GiB used: 223.84 GiB (46.9%)
  ID-1: /dev/nvme0n1 vendor: Kingston model: RBUSNS8154P3512GJ1 size: 476.94 GiB
    speed: 15.8 Gb/s lanes: 2 type: SSD serial: <filter> rev: E8FK12.3 temp: 35.9 C scheme: GPT
Partition:
  ID-1: / size: 467.89 GiB used: 111.92 GiB (23.9%) fs: ext4 dev: /dev/nvme0n1p2
  ID-2: /boot/efi size: 511 MiB used: 6.1 MiB (1.2%) fs: vfat dev: /dev/nvme0n1p1
Swap:
  ID-1: swap-1 type: file size: 2 GiB used: 0 KiB (0.0%) priority: -2 file: /swapfile
USB:
  Hub-1: 1-0:1 info: Hi-speed hub with single TT ports: 8 rev: 2.0 speed: 480 Mb/s
    chip-ID: 1d6b:0002 class-ID: 0900
  Device-1: 1-3:6 info: Logitech Unifying Receiver type: Keyboard,Mouse
    driver: logitech-djreceiver,usbhid interfaces: 2 rev: 2.0 speed: 12 Mb/s power: 98mA
    chip-ID: 046d:c534 class-ID: 0301
  Device-2: 1-8:7 info: Intel Bluetooth wireless interface type: Bluetooth driver: btusb
    interfaces: 2 rev: 2.0 speed: 12 Mb/s power: 100mA chip-ID: 8087:0a2b class-ID: e001
  Hub-2: 2-0:1 info: Super-speed hub ports: 6 rev: 3.1 speed: 10 Gb/s chip-ID: 1d6b:0003
    class-ID: 0900
Sensors:
  System Temperatures: cpu: 27.8 C mobo: N/A
  Fan Speeds (RPM): N/A
Repos:
  Packages: 2482 apt: 2467 flatpak: 15
  No active apt repos in: /etc/apt/sources.list
  Active apt repos in: /etc/apt/sources.list.d/brave-browser-release.list
    1: deb [signed-by=/usr/share/keyrings/brave-browser-archive-keyring.gpg] https: //brave-browser-apt-release.s3.brave.com/ stable main
  Active apt repos in: /etc/apt/sources.list.d/google-chrome.list
    1: deb [arch=amd64] https: //dl.google.com/linux/chrome/deb/ stable main
  Active apt repos in: /etc/apt/sources.list.d/official-package-repositories.list
    1: deb http: //packages.linuxmint.com virginia main upstream import backport
    2: deb http: //archive.ubuntu.com/ubuntu jammy main restricted universe multiverse
    3: deb http: //archive.ubuntu.com/ubuntu jammy-updates main restricted universe multiverse
    4: deb http: //archive.ubuntu.com/ubuntu jammy-backports main restricted universe multiverse
    5: deb http: //security.ubuntu.com/ubuntu/ jammy-security main restricted universe multiverse
  Active apt repos in: /etc/apt/sources.list.d/openrazer-daily-jammy.list
    1: deb [signed-by=/etc/apt/keyrings/openrazer-daily-jammy.gpg] https: //ppa.launchpadcontent.net/openrazer/daily/ubuntu jammy main
  Active apt repos in: /etc/apt/sources.list.d/openrazer-stable-jammy.list
    1: deb [signed-by=/etc/apt/keyrings/openrazer-stable-jammy.gpg] https: //ppa.launchpadcontent.net/openrazer/stable/ubuntu jammy main
  Active apt repos in: /etc/apt/sources.list.d/polychromatic-stable-jammy.list
    1: deb [signed-by=/etc/apt/keyrings/polychromatic-stable-jammy.gpg] https: //ppa.launchpadcontent.net/polychromatic/stable/ubuntu jammy main
  Active apt repos in: /etc/apt/sources.list.d/spotify.list
    1: deb http: //repository.spotify.com stable non-free
  No active apt repos in: /etc/apt/sources.list.d/steam-beta.list
  Active apt repos in: /etc/apt/sources.list.d/steam-stable.list
    1: deb [arch=amd64,i386 signed-by=/usr/share/keyrings/steam.gpg] https: //repo.steampowered.com/steam/ stable steam
    2: deb-src [arch=amd64,i386 signed-by=/usr/share/keyrings/steam.gpg] https: //repo.steampowered.com/steam/ stable steam
Info:
  Processes: 271 Uptime: 3d 8h 36m wakeups: 573 Memory: 15.4 GiB used: 5.15 GiB (33.4%)
  Init: systemd v: 249 runlevel: 5 Compilers: gcc: 11.4.0 alt: 11/12 Client: Cinnamon v: 6.0.4
  inxi: 3.3.13

**Comment by infirit** [comment]
Assuming both the headset and the speakers are connected properly it may be as simple as selecting the right output in the volume mixer application.

Edit: I just realized you may want to send the same audio source to all connected Bluetooth devices. Let us know if this is what want.

**Comment by DavidicusBellefonte** [comment]
> Assuming both the headset and the speakers are connected properly it may be as simple as selecting the right output in the volume mixer application.
> 
> Edit: I just realized you may want to send the same audio source to all connected Bluetooth devices. Let us know if this is what want.

That is correct, I want to connect multiple speakers to one bluetooth audio output from my host device to simulate home theater setups.

**Comment by infirit** [comment]
This is a bit out of scope for blueman so I'll convert this to a discussion in which I can give you some suggestions.

**Comment by infirit** [comment]
I think mint has moved to pipewire but it's similar under pulseaudio. The basic idea is that you create a virtual output device which you link to multiple physical/wireless devices. See https://gitlab.freedesktop.org/pipewire/pipewire/-/wikis/Virtual-Devices

Here is an example of creating a virtual device
```
pw-cli create-node adapter '{ factory.name=support.null-audio-sink node.name=home-cinema media.class=Audio/Sink object.linger=true monitor.channel-volumes=true }'
```

After that you hook up the monitor of this virtual device to all the bluetooth audio devices.

You can hook things up easily with tools like helvum and qpwgraph. You can also use the command line if you want but I never done that myself.

**Reply by DavidicusBellefonte** [reply]
Thank you and I appreciate your expertise and information! I will look into this tonight after work.

**Comment by infirit** [comment]
I am most definitely no expert :sweat_smile: . I have been playing around with my suggestion and it's a broken. The below approach works much better and is based on the example from here https://docs.pipewire.org/page_module_combine_stream.html

You'll have to modify it to suit your needs.

This config should be placed in `~/.config/pipewire/pipewire.conf.d/homecinema.conf`
```
context.modules = [
{   name = libpipewire-module-combine-stream
    args = {
        combine.mode = sink
        node.name = "home-cinema"
        node.description = "My Home Cinema setup"
        combine.latency-compensate = false
        combine.props = {
            audio.position = [ FL FR ]
        }
        stream.props = {
        }
        stream.rules = [
            {
                matches = [
                    # any of the items in matches needs to match, if one does,
                    # actions are emited.
                    {
                        media.class = "Audio/Sink"
                        node.description = "Starship/Matisse HD Audio Controller Analog Stereo"
                    }
                ]
                actions = {
                    create-stream = {}
                }
            }
            {
                matches = [
                    {
                        media.class = "Audio/Sink"
                        node.name = "bluez_output.00_16_94_38_38_F2.1"
                    }
                ]
                actions = {
                    create-stream = {}
                }
            }
        ]
    }
}
]
```
